pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -549,16 +549,31 @@ func (s *Reconciler) createNetworkInterface(ctx context.Context, nicName string)
 		Name:     nicName,
 		VnetName: azure.GenerateVnetName(s.scope.Cluster.Name),
 	}
-	switch set := s.scope.Machine.ObjectMeta.Labels[v1alpha1.MachineRoleLabel]; set {
-	case v1alpha1.Node:
-		networkInterfaceSpec.SubnetName = azure.GenerateNodeSubnetName(s.scope.Cluster.Name)
-	case v1alpha1.ControlPlane:
-		networkInterfaceSpec.SubnetName = azure.GenerateControlPlaneSubnetName(s.scope.Cluster.Name)
-		networkInterfaceSpec.PublicLoadBalancerName = azure.GeneratePublicLBName(s.scope.Cluster.Name)
-		networkInterfaceSpec.InternalLoadBalancerName = azure.GenerateInternalLBName(s.scope.Cluster.Name)
-		networkInterfaceSpec.NatRule = 0
-	default:
-		return errors.Errorf("unknown value %s for label `set` on machine %s, skipping machine creation", set, s.scope.Machine.Name)
+
+	// If any of the following fields is set, ignore role based setting
+	if s.scope.MachineConfig.Subnet != "" || s.scope.MachineConfig.PublicLoadBalancer != "" || s.scope.MachineConfig.InternalLoadBalancer != "" {
+		if s.scope.MachineConfig.Subnet != "" {
+			networkInterfaceSpec.SubnetName = s.scope.MachineConfig.Subnet
+		}
+		if s.scope.MachineConfig.PublicLoadBalancer != "" {
+			networkInterfaceSpec.PublicLoadBalancerName = s.scope.MachineConfig.PublicLoadBalancer
+			networkInterfaceSpec.NatRule = 0
+		}
+		if s.scope.MachineConfig.InternalLoadBalancer != "" {
+			networkInterfaceSpec.InternalLoadBalancerName = s.scope.MachineConfig.InternalLoadBalancer
+		}
+	} else {
```

Comment:
why do we this fallback?

Reply:
https://github.com/openshift/cluster-api-provider-azure/pull/35#discussion_r289452260
https://github.com/openshift/cluster-api-provider-azure/pull/35#discussion_r289452492

Reply:
so @awesomenix if this only for backward compatibility? I don't think we care much at this de v stage, there's no e2e yet . @ingvagabund can drop all no necessary labels code and we put the counter part PR on installer

Reply:
@enxebre Control plane nodes needs to be behind public and internal load balancer, not sure how to determine that without labels. If there is some other method @ingvagabund  feel free to change this logic

Reply:
ah! i see that the master spec would contain the public load balancer and internal load balancer fields. In that case @ingvagabund i support @enxebre comments and drop this as he mentions its more of dev stage.

it would be better if Natrule is dropped as well